File: Python Server/diff.py
def diff(sa,sb):
	
	a = sa.split(",")
	b = sb.split(",")
	
	#print(a)
	#print(b)
	
	mapa = []
	mapb = []
	
	failcount = 0
	
	
	for j in range(0,len(b)):
		for i in range(0,len(a)):
			if(a[i]==b[j]):
				mapa.append(i)
				mapb.append(j)
	
	mapb.sort()
	#print("MAPS")
	#print(mapa)
	#print(mapb)
	ins = list(b)
	j=0
	for i in range(0,len(b)):
		if(j<len(mapa) and mapb[j]==i):
			ins[i] = mapa[j]
			j+=1
		else:
			ins[i]=b[i]
	
	#print("INS\n", ins)
	result = []
	end = 0
	i=0
	while (i<len(ins)):
		if(type(ins[i]) == str):
			result.append(ins[i])
		else:
			if(i+2<len(ins) and type(ins[i+1]) == int and type(ins[i+2]) == int):
				end = i
				while(end<len(ins)-1 and type(ins[end]) == int and ins[end] == ins[end+1]-1):
					end+=1
				
				if(end-i>=2):
					result.append([ins[i],ins[end]])
					i=end
				else:
					result.append(ins[i])
			else:
				result.append(ins[i])
		i+=1
			
	return (result)

File: nodejs2/diff.js
exports.diff = function(s1,s2){
	
	var a = s1.split(",");
	var b = s2.split(",");
	
	var mapb = [];
	var mapa =[];
	var failcount = 0;
	for(var i = 0;i<a.length;i++){
		for(var j = 0;j<b.length;j++){
			if(a[i]==b[j]){
				mapb.push(j);
				mapa.push(i);
			}
		}
	}
	
	mapb.sort(function(a,b){return a-b;});
	
	var ins = b.slice();

	for(var i=0, j=0;i<b.length;i++){
		if(j<mapa.length && mapb[j]==i){
			ins[i]=mapa[j];
			j++;
		}
	}
	
	var result = [];
	var end = 0;
	var i = 0;
	while(i<ins.length){
		if(typeof ins[i] == "string"){
			result.push(ins[i]);
		}else{
			if(i+2<ins.length && typeof ins[i+1] == "number" && typeof ins[i+2] == "number"){
				end = i;
				
				while(end<ins.length-1 && typeof ins[end] == "number" && ins[end] == ins[end+1]-1){
					end++;
				}
				
				if(end-i>2){
					result.push([ins[i],ins[end]]);
					i=end;
				}else{
					result.push(ins[i]);
				}
				
			}else{
				result.push(ins[i]);
			}
		}
		i++;
	}
	return result;
}